### PR TITLE
fix: harden finding contract decision recovery

### PR DESCRIPTION
**Pull request by nrslib**
## Summary

- make Finding Contract Team Leader validation aggregate independent shape, reference, contract, and evidence issues
- retry only validation failures with normal-to-strict recovery, a shared 30-minute deadline, and an emergency 100-call ceiling
- keep evidence classification, recovery prompts, canonical decision digests, and attempt audit records bounded and consistent
- preserve abort and provider or engine error identity without rerunning completed worker parts

Follow-up to #1092.

## Validation

- npm run build
- npm run lint
- npm test: 8,724 passed, 1 skipped
- npm run test:it: 868 passed, 1 skipped
- npm run test:prompt-evals: passed
- npm run test:e2e:mock: passed
- targeted Finding Contract tests: 164 passed

Provider E2E was intentionally not run.

## Review

Three independent Codex review passes approved the final implementation, covering failure semantics, test boundaries, architecture, and AI retry behavior.

## Summary by CodeRabbit

* **改善**
  * Finding Contract の検証エラーを issue とダイジェストに整理し、複数の問題をまとめて確認できるようにしました。
  * 再試行は復旧モード（通常/厳格）と証拠・計画情報を用いて制御し、キャンセル理由や期限切れも適切に反映します。
  * 復旧用プロンプト生成と証拠スナップショットを強化しました。
* **監査**
  * 判定の再試行試行結果を監査用JSONLに記録できるようにしました。
* **テスト**
  * 判定・検証・復旧・中断のシナリオを拡充し、期待値を更新しました。

### CODE REVIEW

File: src/core/workflow/engine/team-leader-finding-contract-decision-retry.ts [hunk 1]
```diff
@@ -1,58 +1,398 @@
...
+            ...buildAttemptTiming(startedAt, deadlineAt),
+            rejectedDecision,
+            terminationReason: 'emergency_call_limit',
+            terminationError: describeTerminationError(error),
+          });
+          throw new FindingContractDecisionRecoveryExhaustedError(error);
+        }
+        promoteRecoveryMode(state, attempt, error, rejectedDecision.repeatCount);
+      }
     }
```
[the remark this answers is not in the document]
Comment (nrslib) [reply src/core/workflow/engine/team-leader-finding-contract-decision-retry.ts:202]: 対応しました。各attempt内のemitTerminatedへ共通化し、既存の終了理由・error・rejectedDecision・timingを同じ形で通知するよう整理しました。対象テスト19件、build、lintを再実行して成功しています。
